scipy/interpolate/fitpack2.py

```diff
@@ -461,7 +461,7 @@ def derivative(self, n=1):
 
         """
         tck = fitpack.splder(self._eval_args, n)
-        return UnivariateSpline._from_tck(tck, self.ext)
+        return UnivariateSpline._from_tck(tck, ext=0)
```

Comment:
This is not right. First, `ext=0` means "extrapolate", not "zeros". Second, the extrapolation mode should be set to `zeros` if and only if the original extrapolation mode was `const`.